pkg/controller/privatelink/conditions/conditions_test.go

```diff
@@ -0,0 +1,358 @@
+				Message: testMessage,
+			}),
+		),
+		cdExists: true,
+		expectConditions: []hivev1.ClusterDeploymentCondition{{
+			Status:  corev1.ConditionTrue,
+			Type:    hivev1.PrivateLinkFailedClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}, {
+			Status:  corev1.ConditionFalse,
+			Type:    hivev1.PrivateLinkReadyClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}},
+		expectLogs: []string{"setting PrivateLinkFailedClusterDeploymentCondition to true"},
+	}, { // There should be a log message when the ready condition changed to false
+		name: "ready condition changed to false",
+		cd: cdBuilder.Build(
+			testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+				Type:    hivev1.PrivateLinkFailedClusterDeploymentCondition,
+				Status:  corev1.ConditionTrue,
+				Reason:  testReason,
+				Message: testMessage,
+			}),
+			testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+				Type:    hivev1.PrivateLinkReadyClusterDeploymentCondition,
+				Status:  corev1.ConditionTrue,
+				Reason:  testReason,
+				Message: testMessage,
+			}),
+		),
+		expectConditions: []hivev1.ClusterDeploymentCondition{{
+			Status:  corev1.ConditionTrue,
+			Type:    hivev1.PrivateLinkFailedClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}, {
+			Status:  corev1.ConditionFalse,
+			Type:    hivev1.PrivateLinkReadyClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}},
+		cdExists:   true,
+		expectLogs: []string{"setting PrivateLinkFailedClusterDeploymentCondition to true"},
+	}, { // There should be no errors when no conditions change
+		name: "no change",
+		cd: cdBuilder.Build(
+			testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+				Type:    hivev1.PrivateLinkFailedClusterDeploymentCondition,
+				Status:  corev1.ConditionTrue,
+				Reason:  testReason,
+				Message: testMessage,
+			}),
+			testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+				Type:    hivev1.PrivateLinkReadyClusterDeploymentCondition,
+				Status:  corev1.ConditionFalse,
+				Reason:  testReason,
+				Message: testMessage,
+			}),
+		),
+		cdExists: true,
+		expectConditions: []hivev1.ClusterDeploymentCondition{{
+			Status:  corev1.ConditionTrue,
+			Type:    hivev1.PrivateLinkFailedClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}, {
+			Status:  corev1.ConditionFalse,
+			Type:    hivev1.PrivateLinkReadyClusterDeploymentCondition,
+			Reason:  testReason,
+			Message: testMessage,
+		}},
+	}}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			existingCD := &hivev1.ClusterDeployment{}
+			if test.cdExists {
+				existingCD = test.cd
+			}
+			fakeClient := client.Client(testfake.NewFakeClientBuilder().WithRuntimeObjects(existingCD).Build())
+
+			logger, hook := testlogger.NewLoggerWithHook()
+
+			err := SetErrCondition(fakeClient, test.cd, testReason, errors.New(testMessage), logger)
```

Comment:
Oh yeah, that was an artifact of adding `SetErrConditionWithRetry` after the fact. It could absolutely be private, but I'm trying to avoid changing functional code in this PR.

Reply:
Hopefully this discussion becomes moot once https://issues.redhat.com/browse/HIVE-2686 is addressed.